smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -160,6 +160,93 @@ SlangBasicTranslationTest >> testBlockValueAssignmentIntoSameVariable [
 }'
 ]
 
+{ #category : #'tests-blocks' }
+SlangBasicTranslationTest >> testBlockValueWithConstantArgument [
+
+	"Case:
+	
+	[ :i | a := i ] value: 7"
+
+	| translation |
+
+	translation := self translate: (TSendNode new
+		setSelector: #value:
+		receiver: (TStmtListNode new
+			setArguments: #(i)
+			statements: {
+				TAssignmentNode new setVariable: (TVariableNode new setName: 'a') expression: (TVariableNode new setName: 'i')
+			})
+		arguments: { TConstantNode value: 7 }).
+
+	self assert: translation trimBoth equals: '{
+	a = 7;
+}'
+]
+
+{ #category : #'tests-blocks' }
+SlangBasicTranslationTest >> testBlockValueWithNonLeafArgument [
+
+	"Case:
+	
+	[ :i | a := i ] value: 7 + 5"
+
+	| translation |
+	translation := self translate: (TSendNode new
+			                setSelector: #value:
+			                receiver: (TStmtListNode new
+					                 setArguments: #( i )
+					                 statements: { (TAssignmentNode new
+							                  setVariable: (TVariableNode new setName: 'a')
+							                  expression: (TVariableNode new setName: 'i')) })
+			                arguments: { (TSendNode new
+					                 setSelector: #+
+					                 receiver: (TConstantNode value: 7)
+					                 arguments: { (TConstantNode value: 5) }) }).
+
+	self assert: translation trimBoth equals: 'i = 7 + 5;
+{
+	a = i;
```

Comment:
This means we should test (maybe later) that the variable `i` has been hoisted in the method, right?